pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -370,17 +370,21 @@ protected CompletableFuture<Optional<TopicPolicies>> getTopicPoliciesAsyncWithRe
     }
 
     protected boolean checkBacklogQuota(BacklogQuota quota, RetentionPolicies retention) {
-        if (retention == null || retention.getRetentionSizeInMB() <= 0 || retention.getRetentionTimeInMinutes() <= 0) {
+        if (retention == null
+                || (retention.getRetentionSizeInMB() <= 0 && retention.getRetentionTimeInMinutes() <= 0)) {
```

Comment:
Looks like the added test can't cover this branch?

Reply:
@codelipenghui As mentioned in the issue. The test in L3417 won't fail before this change.

Reply:
And the case of `(retention.getRetentionSizeInMB() <= 0 && retention.getRetentionTimeInMinutes() <= 0)` is the default setting, it's covered by previous unit tests.